Docs Reorg (#53)
waku-org/waku.guide: docs/getting-started/reference/glossary.md

---
title: Glossary
---

# Waku Docs Glossary

Definitions and usage of the terminology used in the Waku ecosystem.

### Bootstrapping

> Bootstrapping is the initial [peer discovery](#peer-discovery) process where new [nodes](#node) on the network locate and establish connections with one another.

Reviewer: Information about bootstrap nodes must be provided out-of-band (See ... discovery section) as they are the initial entry point to the Waku network.
Once connected, other peer discovery methods can be used.

### Content Topic

> A content topic is a string attached to [messages](#waku-message) to enable [protocol-level](#protocol) features like selective message processing and retrieval based on specific criteria.

Reviewer: Link to RFC or doc section or both?

### Dappnode

> Dappnode is an open-source platform that simplifies the hosting and management of decentralized applications and blockchain nodes, including [Waku](#waku).

Reviewer: Link to their website?

### Discv5

> Discv5 is a [peer discovery](#peer-discovery) mechanism using a Distributed Hash Table (DHT) to store [ENR](#enr) records, providing censorship resistance, load distribution, and enhanced network resilience.

Reviewer: Link to RFC or doc section or both?

### DNS Discovery

> DNS discovery is a [peer discovery](#peer-discovery) mechanism that allows the registration of an [ENR](#enr) tree in the TXT field of a domain name, enabling the storage of [node](#node) connection details and promoting decentralization.

Reviewer: ```suggestion
DNS discovery is a [peer discovery](#peer-discovery) mechanism that allows the retrieval of an [ENR](#enr) tree from the TXT field of a domain name, enabling the storage of [node](#node) connection details and promoting decentralization.
```

### ENR

> Ethereum Node Record (ENR) is a specification used to represent and identify [nodes](#node), facilitating [discovery](#peer-discovery) and communication within the network.

Reviewer: In addition to connection details, Waku ENR contains information about the node configuration such as enabled protocol and shards.

### Filter

> Filter is a [protocol](#protocol) that enables [light nodes](#light-node) to selectively subscribe to specific [messages](#waku-message) transmitted by [peers](#peer) using [content topics](#content-topic).

Reviewer: I would mention that this is an cheap access or light access to the Relay Network.

### GossipSub

> Gossipsub is a [protocol](#protocol) for efficient and scalable information dissemination in decentralized networks commonly used in blockchain systems.

Reviewer: ```suggestion
GossipSub is a [protocol](#protocol) for efficient and scalable information dissemination in decentralized networks commonly used in blockchain systems.
```

### Light Node

> A light node is a [resource-limited](#resource-limited) device or client that selectively subscribes to specific [messages](#waku-message) transmitted by other [peers](#peer) on the network.

Reviewer: ```suggestion
A light node is a [resource-limited](#resource-limited) device or client that leverage service nodes to access the Waku Relay network.
```

### Light Push

Light push is a protocol enabling [light nodes](#light-node) to send [messages](#waku-message) to the [Relay](#relay) network and receive acknowledgments confirming that a [peer](#peer) has received them.

### Node

> A node is a device or client that plays an active role in the operation of the [Waku network](#waku), participating to enable secure and private peer-to-peer Web3 communication.

Reviewer: hum, a light node would not play much of an "active" role. Maybe go for the angle of implementing Waku protocols and leveraging Waku network.

### Noise

> Noise is the integration of the Noise Protocol Framework into the [Waku network](#waku) to enable secure key-exchange mechanisms for encrypted communication with confidentiality, authenticity, integrity, strong forward secrecy, and identity-hiding properties.

Reviewer: Well 

> Noise is a framework for building crypto protocols.

http://noiseprotocol.org/

Then you can mention that Waku Noise https://rfc.vac.dev/spec/35/ is a specified way to use Noise over Waku

### Payload

> Payload provides guidelines for implementing secure and private communication in the network. It covers encryption, decryption, and signing methods for message payloads, focusing on confidentiality, authenticity, integrity, and unlinkability.

Reviewer: Hum... You are describing the https://rfc.vac.dev/spec/26/ protocol. I think it's confusing. I'd define two things:

### Payload

The payload field of a Waku Message is application data. This is the business logic message that applications are transmitting between clients over Waku. The payload can be encrypted by the application or using methods specified in Waku Message Payload Encryption.

#### Waku Message Payload Encryption

... provide guidelines for implementing secure and private communication in the network. It covers encryption, decryption, and signing methods for message payloads, focusing on confidentiality, authenticity, integrity, and unlinkability.

### Peer

A peer refers to other [nodes](#node) and participants of the [Waku network](#waku) with whom communication and interaction are possible.

### Peer Discovery

Peer discovery is the process where a [node](#node) locates and connects with [peers](#peer) to establish communication and exchange information.

### Peer Exchange

Peer exchange is a [peer discovery](#peer-discovery) mechanism that enables [light nodes](#light-node) to request and receive peers from other nodes in the network, allowing them to bootstrap and expand their connections without depending on [Discv5](#discv5).

### Protocol

A protocol is a set of rules that enables [nodes](#node) within the [Waku network](#waku) to perform various functionalities such as message sending, relaying, filtering, storing, retrieving, and more.

### Pub/Sub

> Publish/Subscribe (Pub/Sub) is a messaging pattern where publishers distribute messages to subscribers without the direct knowledge of recipients, allowing efficient one-to-many communication.

Reviewer: > without the direct knowledge of recipients

What are you trying to say?

### Pub/Sub Topic

> A Pub/Sub topic is a routing mechanism for messages, indicating a subscription to receive messages on that topic for relay purposes.

Reviewer: Huuum

it's a string to identify the topic of interested in gossipsub peers. peers interested in the same topic are likely to maintain a connection and forward messages they received on said topic.

### Rate Limit Nullifiers

Rate Limit Nullifiers (RLN) are a construct based on zero-knowledge proofs that provide an anonymous rate-limited messaging framework, preserving message owner anonymity while preventing spam or DoS attacks.

### Relay

Relay is a [protocol](#protocol) that extends the [GossipSub protocol](#gossipsub) to enable secure and censorship-resistant [message](#waku-message) dissemination among [peers](#peer) while preserving privacy.

### Resource-Limited

> Resource-limited refers to environments or devices restricting available resources, including bandwidth, CPU, memory, disk, and battery power.

Reviewer: add "mostly offline" in the keywords and maybe good to add example of browser and mobile phone.

### RLN Relay

RLN Relay is an extension of the [Relay protocol](#relay) that uses [Rate Limit Nullifiers (RLN)](#rate-limit-nullifiers) to prevent spam economically by enforcing bandwidth caps, imposing penalties, and facilitating network removal for spammers.

### SDK

SDKs are tools, libraries, and resources to integrate Waku's private, secure, and censorship-free communication features into various applications.

### Store

Store is a [protocol](#protocol) that enables the storage of relayed [messages](#waku-message) in the network, allowing offline peers to retrieve missed messages upon reconnecting to the network.

### Transport

A transport is a network mechanism that establishes connections between [peers](#peer) and enables efficient transmission, routing, and delivery of data packets.

### Waku

> Waku is a family of private, secure, decentralized, and peer-to-peer Web3 communication [protocols](#protocol) designed to operate in [resource-limited](#resource-limited) environments and suitable for [node](#node) or desktop application use.

Reviewer: Maybe mention Waku Network here or have an entry for it.
